tests/functional/event_handler/test_api_middlewares.py

```diff
@@ -397,6 +397,46 @@ def dummy_route():
     assert result["statusCode"] == 200
 
 
+@pytest.mark.parametrize(
+    "app, event",
+    [
+        (ApiGatewayResolver(proxy_type=ProxyEventType.APIGatewayProxyEvent), API_REST_EVENT),
+        (APIGatewayRestResolver(), API_REST_EVENT),
+        (APIGatewayHttpResolver(), API_RESTV2_EVENT),
+    ],
+)
+def test_api_gateway_middleware_with_include_router_prefix(app: EventHandlerInstance, event):
+    # GIVEN a router instance
+    router = Router()
+
+    def app_middleware(app: EventHandlerInstance, next_middleware: NextMiddleware):
+        # inject a variable into resolver context
+        app.append_context(injected="injected_value")
+        response = next_middleware(app)
+
+        return response
```

Comment:
```suggestion
        return next_middleware(app)
```

suggestion: we can shorten the test since we're not altering the response.